common/src/Model/Brush.cpp

```diff
@@ -1064,11 +1053,98 @@ namespace TrenchBroom {
             }
 
             const PolyhedronMatcher<BrushGeometry> matcher(*m_geometry, newGeometry, vertexMapping);
-            doSetNewGeometry(worldBounds, matcher, newGeometry);
+            doSetNewGeometry(worldBounds, matcher, newGeometry, uvLock);
         }
 
-        void Brush::doSetNewGeometry(const vm::bbox3& worldBounds, const PolyhedronMatcher<BrushGeometry>& matcher, BrushGeometry& newGeometry) {
-            matcher.processRightFaces(FaceMatchingCallback());
+        std::tuple<bool, vm::mat4x4> Brush::findTransformForUVLock(const PolyhedronMatcher<BrushGeometry>& matcher, BrushFaceGeometry* left, BrushFaceGeometry* right) {
```

Comment:
Could you please add a little bit of documentation so that someone can understand what the function is supposed to do? Just a sentence or two should be enough.

Reply:
This one has documentation, but in the header.
Should I move it to the .cpp for private helper functions like this?

Also, I'll add docs for `applyUVLock` which is a bit confusing.

Reply:
No, that’s fine! I missed it, sorry.